### PR TITLE
nixos/systemd-boot: fix tests and upgrade issues

### LINKED ISSUE #151336: systemd-boot update touches EFI variables when told not to
### Describe the bug
During systemd-boot update step `bootctl` attempts to touch EFI variables, even when `boot.loader.efi.canTouchEfiVariables` is set to `false`.

### Steps To Reproduce
1. Install NixOS 21.05 with systemd-boot and `boot.loader.efi.canTouchEfiVariables = false;`
2. Upgrade to NixOS 21.11 (fa8aa7ff5995d539aabc6f6a009dfcc1e3fd33a8)
3. Configuration switch fails with:
```
Copied "/nix/store/rgjrn2vl55pyxahfggvbdgl5g0xn0an1-systemd-249.5/lib/systemd/boot/efi/systemd-bootx64.efi" to "/boot/EFI/systemd/systemd-bootx64.efi".
Copied "/nix/store/rgjrn2vl55pyxahfggvbdgl5g0xn0an1-systemd-249.5/lib/systemd/boot/efi/systemd-bootx64.efi" to "/boot/EFI/BOOT/BOOTX64.EFI".
updating systemd-boot from (247) to (249.5)
Failed to create EFI Boot variable entry: Input/output error
Traceback (most recent call last):
  File "/nix/store/5f69p1syxlsfyxxb54zs1gc59ngf2cxz-systemd-boot", line 314, in <module>
    main()
  File "/nix/store/5f69p1syxlsfyxxb54zs1gc59ngf2cxz-systemd-boot", line 267, in main
    subprocess.check_call(["/nix/store/rgjrn2vl55pyxahfggvbdgl5g0xn0an1-systemd-249.5/bin/bootctl", "--path=/boot", "update"])
  File "/nix/store/wl02plhc6zf84m6x9984l42wnnnbly5m-python3-3.9.6/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/rgjrn2vl55pyxahfggvbdgl5g0xn0an1-systemd-249.5/bin/bootctl', '--path=/boot', 'update']' returned non-zero exit status 1.
```
An error is just an indication of update trying to touch EFI variables. (I have EFI vars touching disabled on that system because of some unrelated firmware breakage)
Manually running `bootctl --path=/boot update --no-variables` properly finishes, further switches work fine.

### Expected behavior
systemd-boot update should not touch EFI variables during upgrade.

### Notify maintainers
@andir @flokli @kloenk

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.64, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by tpwrules**
###### Description of changes

Fixes several issues around systemd-boot's tests and upgrades. See commit messages for details.

Fixes https://github.com/NixOS/nixpkgs/issues/151336

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ncfavier — APPROVED** [review]

**Comment by ncfavier** [comment]
@ofborg test systemd-boot
@ofborg test installer.simpleUefiSystemdBoot

**Comment by tpwrules** [comment]
Strange that the aarch64-linux tests timed out. Is it being double-emulated by qemu?

File: nixos/modules/virtualisation/qemu-vm.nix [hunk 1]
```diff
@@ -211,7 +211,7 @@ let
             ''
               mkdir $out
               diskImage=$out/disk.img
-              ${qemu}/bin/qemu-img create -f qcow2 $diskImage "60M"
+              ${qemu}/bin/qemu-img create -f qcow2 $diskImage "120M"
```
Comment (RaitoBezarius) [inline nixos/modules/virtualisation/qemu-vm.nix:214]: Do we really need to double the size of the ESP partition?

Comment (RaitoBezarius) [reply nixos/modules/virtualisation/qemu-vm.nix:214]: Okay, aarch64 stuff.

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**